mesh/mesh.go

```diff
@@ -641,45 +641,56 @@ func (msh *Mesh) AddTXsFromProposal(ctx context.Context, layerID types.LayerID,
 
 // AddBallot to the mesh.
 func (msh *Mesh) AddBallot(ballot *types.Ballot) error {
-	if err := msh.addBallot(ballot); err != nil {
-		return err
+	decoded, err := msh.trtl.DecodeBallot(ballot)
+	if err != nil {
+		return fmt.Errorf("decode ballot %s: %w", ballot.ID(), err)
 	}
-	msh.trtl.OnBallot(ballot)
-	return nil
-}
-
-func (msh *Mesh) addBallot(b *types.Ballot) error {
-	mal, err := identities.IsMalicious(msh.cdb, b.SmesherID().Bytes())
+	if decoded == nil {
```

Comment:
what are the conditions that cause decoded to return nil here? please also add comment in the interface in system pkg